nvmeadm/tests/discovery_test.rs

```diff
@@ -1,7 +1,147 @@
 use nvmeadm::nvmf_discovery::{disconnect, DiscoveryBuilder};
 
+use failure::Error;
+use std::{
+    fs::File,
+    io::prelude::*,
+    net::{SocketAddr, TcpStream},
+    path::Path,
+    process::Command,
+    thread,
+    time::Duration,
+};
+
+static CONFIG_TEXT: &str = "[Malloc]
+  NumberOfLuns 1
+  LunSizeInMB  64
+  BlockSize    4096
+[Nvmf]
+  AcceptorPollRate 10000
+  ConnectionScheduler RoundRobin
+[Transport]
+  Type TCP
+  # reduce memory requirements
+  NumSharedBuffers 32
+[Subsystem1]
+  NQN nqn.2019-05.io.openebs:disk2
+  Listen TCP 127.0.0.1:NVMF_PORT
+  AllowAnyHost Yes
+  SN MAYASTOR0000000001
+  MN NEXUSController1
+  MaxNamespaces 1
+  Namespace Malloc0 1
+# although not used we still have to reduce mem requirements for iSCSI
+[iSCSI]
+  MaxSessions 1
+  MaxConnectionsPerSession 1
+";
+
+static CONFIG_FILE: &str = "/tmp/nvmeadm_nvmf_target.config";
+
+static SERVED_DISK_NQN: &str = "nqn.2019-05.io.openebs:disk2";
+
+static TARGET_PORT: u32 = 9523;
+
+// Writes out a config file for spdk, but with the specified port for nvmf to
+// use
+fn create_config_file(config_file: &str, nvmf_port: &str) {
+    let path = Path::new(config_file);
+    let mut config = match File::create(&path) {
+        Err(reason) => panic!(
+            "Unable to create {}: {}",
+            path.display(),
+            reason.to_string()
+        ),
+        Ok(config) => config,
+    };
+
+    let after = CONFIG_TEXT.replace("NVMF_PORT", nvmf_port);
+
+    match config.write_all(after.as_bytes()) {
+        Err(reason) => panic!(
+            "Unable to write to {}: {}",
+            path.display(),
+            reason.to_string()
+        ),
+        Ok(_) => println!("Wrote to file"),
```

Comment:
did you mean to leave this print?

Reply:
Yes ... it's all hidden normally but I found it useful to help follow what's happening with --nocapture.

Reply:
A bunch of static have become const